test/letters/reducers/index.unit.spec.js

```diff
@@ -157,68 +167,183 @@ describe('letters reducer', () => {
+    expect(state.states).to.equal(states);
+    expect(state.statesAvailable).to.be.true;
   });
 
-  it('should handle failure to fetch states', () => {
-    const state = lettersReducer.letters(
-      initialState,
-      { type: 'GET_ADDRESS_STATES_FAILURE' }
+  it('should handle empty response for the states', () => {
+    const state = reduce(
+      {
+        type: GET_ADDRESS_STATES_SUCCESS,
+        states: {
+          data: {
+            attributes: {
+              states: []
+            }
+          }
+        }
+      }
     );
 
+    expect(state.states).to.not.contain('IL');
+    expect(state.statesAvailable).to.be.false;
+  });
+
+  it('should handle failure to fetch states', () => {
+    const state = reduce({ type: GET_ADDRESS_STATES_FAILURE });
+
     expect(state.states).to.be.empty;
+    expect(state.statesAvailable).to.be.false;
+  });
+
+
+  it('should handle a a letter eligibility error', () => {
+    const state = reduce({ type: LETTER_ELIGIBILITY_ERROR });
+
+    expect(state.lettersAvailability).to.equal(AVAILABILITY_STATUSES.letterEligibilityError);
+  });
+
+  it('should handle updating a benefit summary request option', () => {
+    const state = reduce(
+      {
+        type: UPDATE_BENFIT_SUMMARY_REQUEST_OPTION,
+        propertyPath: 'foo',
+        value: 'bar'
+      }
+    );
+
+    expect(state.requestOptions.foo).to.equal('bar');
+  });
+
+  it('should handle downloading a pdf', () => {
+    const state = reduce(
+      {
+        type: GET_LETTER_PDF_DOWNLOADING,
+        data: 'foo' // The letter name
+      }
+    );
+
+    expect(state.letterDownloadStatus.foo).to.equal(DOWNLOAD_STATUSES.downloading);
+  });
+
+  it('should handle successfully downloading a pdf', () => {
+    const state = reduce(
+      {
+        type: GET_LETTER_PDF_SUCCESS,
+        data: 'foo' // The letter name
+      }
+    );
+
+    expect(state.letterDownloadStatus.foo).to.equal(DOWNLOAD_STATUSES.success);
+  });
+
+  it('should handle failing to download a pdf', () => {
+    const state = reduce(
+      {
+        type: GET_LETTER_PDF_FAILURE,
+        data: 'foo' // The letter name
+      }
+    );
+
+    expect(state.letterDownloadStatus.foo).to.equal(DOWNLOAD_STATUSES.failure);
+  });
+
+  it('should handle save address pending', () => {
+    const state = reduce({ type: SAVE_ADDRESS_PENDING });
+
+    expect(state.savePending).to.be.true;
+  });
+
+  it('should handle successfully saving an address', () => {
+    const state = reduce(
+      {
+        type: SAVE_ADDRESS_SUCCESS,
+        address: '123 Main St'
+      }
+    );
+
+    expect(state.address).to.equal('123 Main St');
+    expect(state.savePending).to.be.false;
+  });
+
+  it('should handle failing to save an address', () => {
+    const state = reduce(
+      {
+        type: SAVE_ADDRESS_FAILURE,
+        address: '123 Main St'
+      }
+    );
+
+    expect(state.savePending).to.be.false;
+    expect(state.saveAddressError).to.be.true;
```

Comment:
This is interesting, I hadn't noticed before that the code worked this way. I wonder why we storing 2 separate fields here instead of doing what we do for fetching letters and address, and store one field and have the values be something like: `pending`, `failure`, `success`, etc?

Reply:
Yeah... 🤔  Now that you mention it, that is odd. Let's write up a ticket to address that.